lib/ansible/plugins/action/script.py

```diff
@@ -55,6 +55,10 @@ def run(self, tmp=None, task_vars=None):
                 self._remove_tmp_path(tmp)
                 return dict(skipped=True, msg=("skipped, since %s does not exist" % removes))
 
+        chdir = self._task.args.get('chdir')
+        if chdir:
+            chdir = os.path.abspath(chdir)
```

Comment:
I don't agree with you, This actually works.
`os.path.abspath()` does not rely on *real* filesystem, it just performs path simplification and works on strings.
You may refer to `os.path.realpath()`, that does perform an os call to check real filesystem.

But you are right on one point: I have not tested the case when the remote node is a Windows server. I will check the Python API.